-1

**Failed Tests**:   Build 
**Size**: This PR adds an extra 60KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-92ca8b/55084/summary.html
**COMMIT**: 9cf73d05d137c4aa10a269fab25ddc07d3b7f04c
**CMSSW**: CMSSW_20_1_X_2026-08-04-1100/el9_amd64_gcc13
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/51626/55084/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Failed Build

I found compilation error when building: 

<pre>>> Compiling edm plugin src/PerfTools/AllocMonitor/plugins/moduleAlloc_setupFile.cc
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02953/el9_amd64_gcc13/external/gcc/13.4.0-6908cfdf803923e783448096ca4f0923/bin/c++ -c -DCMS_MICRO_ARCH='x86-64-v3' -DGNU_GCC -D_GNU_SOURCE -DTBB_USE_GLIBCXX_VERSION=130400 -DTBB_SUPPRESS_DEPRECATED_MESSAGES -DTBB_PREVIEW_RESUMABLE_TASKS=1 -DTBB_PREVIEW_TASK_GROUP_EXTENSIONS=1 -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DBOOST_MATH_DISABLE_STD_FPCLASSIFY -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX -DBOOST_MPL_IGNORE_PARENTHESES_WARNING -DCMSSW_GIT_HASH='CMSSW_20_1_X_2026-08-04-1100' -DPROJECT_NAME='CMSSW' -DPROJECT_VERSION='CMSSW_20_1_X_2026-08-04-1100' -Isrc -Ipoison -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02953/el9_amd64_gcc13/cms/cmssw-patch/CMSSW_20_1_X_2026-08-04-1100/src -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02953/el9_amd64_gcc13/external/pcre/8.43-7f0edb13c32a66555b25b5433dd2ceed/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02953/el9_amd64_gcc13/external/boost/1.91.0-12be15fc9ce66d33c35e60e22602d5c9/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02953/el9_amd64_gcc13/external/bz2lib/1.0.8-353e2ef1c2f1d06b28a8938defa197bc/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02953/el9_amd64_gcc13/external/libuuid/2.40-d94705b84648fd294d63bf9de23bc03b/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02953/el9_amd64_gcc13/lcg/root/6.36.13-13eefd63b87f249157ea6e9e39659b69/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02953/el9_amd64_gcc13/external/tbb/v2022.3.0-4e3eb7de4a12442dd626d6802d854695/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02953/el9_amd64_gcc13/external/xz/5.8.3-3303bcf2cc1251389e9225885f971ea3/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02953/el9_amd64_gcc13/external/zlib/1.3.2-53c061acc706fe8f28fd9477d9981c20/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02953/el9_amd64_gcc13/external/md5/1.0.0-26057075013e190e56dad37d35219376/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02953/el9_amd64_gcc13/external/tinyxml2/6.2.0-da2efb8ba86adb2f29c708e411784949/include -O3 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++20 -ftree-vectorize -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -Wno-error=array-bounds -Warray-bounds -fuse-ld=bfd -march=x86-64-v3 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-unused-parameter -Wunused -Wparentheses -Werror=return-type -Werror=unused-value -Werror=unused-result -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -Wno-error=unused-variable -DBOOST_DISABLE_ASSERTS -flto=auto -fipa-icf -flto-odr-type-merging -fno-fat-lto-objects -Wodr -fPIC -MMD -MF tmp/el9_amd64_gcc13/src/PerfTools/AllocMonitor/plugins/PerfToolsAllocMonitorAuto/moduleAlloc_setupFile.cc.d src/PerfTools/AllocMonitor/plugins/moduleAlloc_setupFile.cc -o tmp/el9_amd64_gcc13/src/PerfTools/AllocMonitor/plugins/PerfToolsAllocMonitorAuto/moduleAlloc_setupFile.cc.o
>> Compiling edm plugin src/PerfTools/AllocMonitor/plugins/monitor_file_utilities.cc
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02953/el9_amd64_gcc13/external/gcc/13.4.0-6908cfdf803923e783448096ca4f0923/bin/c++ -c -DCMS_MICRO_ARCH='x86-64-v3' -DGNU_GCC -D_GNU_SOURCE -DTBB_USE_GLIBCXX_VERSION=130400 -DTBB_SUPPRESS_DEPRECATED_MESSAGES -DTBB_PREVIEW_RESUMABLE_TASKS=1 -DTBB_PREVIEW_TASK_GROUP_EXTENSIONS=1 -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DBOOST_MATH_DISABLE_STD_FPCLASSIFY -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX -DBOOST_MPL_IGNORE_PARENTHESES_WARNING -DCMSSW_GIT_HASH='CMSSW_20_1_X_2026-08-04-1100' -DPROJECT_NAME='CMSSW' -DPROJECT_VERSION='CMSSW_20_1_X_2026-08-04-1100' -Isrc -Ipoison -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02953/el9_amd64_gcc13/cms/cmssw-patch/CMSSW_20_1_X_2026-08-04-1100/src -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02953/el9_amd64_gcc13/external/pcre/8.43-7f0edb13c32a66555b25b5433dd2ceed/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02953/el9_amd64_gcc13/external/boost/1.91.0-12be15fc9ce66d33c35e60e22602d5c9/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02953/el9_amd64_gcc13/external/bz2lib/1.0.8-353e2ef1c2f1d06b28a8938defa197bc/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02953/el9_amd64_gcc13/external/libuuid/2.40-d94705b84648fd294d63bf9de23bc03b/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02953/el9_amd64_gcc13/lcg/root/6.36.13-13eefd63b87f249157ea6e9e39659b69/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02953/el9_amd64_gcc13/external/tbb/v2022.3.0-4e3eb7de4a12442dd626d6802d854695/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02953/el9_amd64_gcc13/external/xz/5.8.3-3303bcf2cc1251389e9225885f971ea3/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02953/el9_amd64_gcc13/external/zlib/1.3.2-53c061acc706fe8f28fd9477d9981c20/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02953/el9_amd64_gcc13/external/md5/1.0.0-26057075013e190e56dad37d35219376/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02953/el9_amd64_gcc13/external/tinyxml2/6.2.0-da2efb8ba86adb2f29c708e411784949/include -O3 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++20 -ftree-vectorize -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -Wno-error=array-bounds -Warray-bounds -fuse-ld=bfd -march=x86-64-v3 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-unused-parameter -Wunused -Wparentheses -Werror=return-type -Werror=unused-value -Werror=unused-result -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -Wno-error=unused-variable -DBOOST_DISABLE_ASSERTS -flto=auto -fipa-icf -flto-odr-type-merging -fno-fat-lto-objects -Wodr -fPIC -MMD -MF tmp/el9_amd64_gcc13/src/PerfTools/AllocMonitor/plugins/PerfToolsAllocMonitorAuto/monitor_file_utilities.cc.d src/PerfTools/AllocMonitor/plugins/monitor_file_utilities.cc -o tmp/el9_amd64_gcc13/src/PerfTools/AllocMonitor/plugins/PerfToolsAllocMonitorAuto/monitor_file_utilities.cc.o
src/PerfTools/AllocMonitor/plugins/ModuleAllocMonitor.cc: In constructor 'ModuleAllocMonitor::ModuleAllocMonitor(const edm::ParameterSet&, edm::ActivityRegistry&)':
src/PerfTools/AllocMonitor/plugins/ModuleAllocMonitor.cc:164:11: error: 'class edm::ActivityRegistry' has no member named 'watchPostESModuleRegistration'; did you mean 'watchPostESModuleConstruction'?
  164 |       iAR.watchPostESModuleRegistration([this](auto const& iDescription) {
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |           watchPostESModuleConstruction
src/PerfTools/AllocMonitor/plugins/PhaseAllocMonitor.cc: In constructor 'PhaseAllocMonitor::PhaseAllocMonitor(const edm::ParameterSet&, edm::ActivityRegistry&)':
src/PerfTools/AllocMonitor/plugins/PhaseAllocMonitor.cc:114:9: error: 'class edm::ActivityRegistry' has no member named 'watchJobFailure'
</pre>

--- a/EventFilter/Utilities/interface/FastMonitoringService.h
+++ b/EventFilter/Utilities/interface/FastMonitoringService.h
@@ -178,7 +178,6 @@ namespace evf {
     std::string makeInputLegendaJson();
 
     void preallocate(edm::service::SystemBounds const&);
-    void jobFailure();
     void preBeginJob(edm::ProcessContext const& pc);
 
     void preModuleBeginJob(edm::ModuleDescription const&);

--- a/EventFilter/Utilities/src/FastMonitoringService.cc
+++ b/EventFilter/Utilities/src/FastMonitoringService.cc
@@ -206,7 +206,6 @@ namespace evf {
         totalEventsProcessed_(0),
         verbose_(iPS.getUntrackedParameter<bool>("verbose")) {
     reg.watchPreallocate(this, &FastMonitoringService::preallocate);  //receiving information on number of threads
-    reg.watchJobFailure(this, &FastMonitoringService::jobFailure);    //global
 
     reg.watchPreBeginJob(this, &FastMonitoringService::preBeginJob);
     reg.watchPreModuleBeginJob(this, &FastMonitoringService::preModuleBeginJob);  //global
@@ -468,8 +467,6 @@ namespace evf {
     }
     return false;
   }
-
-  void FastMonitoringService::jobFailure() { fmt_->m_data.macrostate_ = FastMonState::sError; }
 
   //new output module name is stream
   void FastMonitoringService::preModuleBeginJob(const edm::ModuleDescription& desc) {

--- a/FWCore/Framework/src/EventSetupProvider.cc
+++ b/FWCore/Framework/src/EventSetupProvider.cc
@@ -93,9 +93,6 @@ namespace edm {
     void EventSetupProvider::add(std::shared_ptr<ESProductResolverProvider> iProvider) {
       assert(iProvider.get() != nullptr);
       dataProviders_->push_back(iProvider);
-      if (activityRegistry_) {
-        activityRegistry_->postESModuleRegistrationSignal_.emit(iProvider->description());
-      }
     }
 
     using RecordProviders = std::vector<std::shared_ptr<EventSetupRecordProvider>>;

--- a/FWCore/MessageService/plugins/MessageLogger.cc
+++ b/FWCore/MessageService/plugins/MessageLogger.cc
@@ -223,7 +223,6 @@ namespace edm {
       iRegistry.watchPostBeginJob(this, &MessageLogger::postBeginJob);
       iRegistry.watchPreEndJob(this, &MessageLogger::preEndJob);
       iRegistry.watchPostEndJob(this, &MessageLogger::postEndJob);
-      iRegistry.watchJobFailure(this, &MessageLogger::jobFailure);  // change log 14
 
       iRegistry.watchPreModuleConstruction(this, &MessageLogger::preModuleConstruction);
       iRegistry.watchPostModuleConstruction(this, &MessageLogger::postModuleConstruction);
@@ -973,13 +972,6 @@ namespace edm {
     }
 
     void MessageLogger::postEndJob() {
-      summarizeInJobReport();    // Put summary info into Job Rep  // change log 10
-      MessageLoggerQ::MLqSUM();  // trigger summary info.		// change log 9
-    }
-
-    void MessageLogger::jobFailure() {
-      MessageDrop* messageDrop = MessageDrop::instance();
-      messageDrop->setSinglet("jobFailure");
       summarizeInJobReport();    // Put summary info into Job Rep  // change log 10
       MessageLoggerQ::MLqSUM();  // trigger summary info.		// change log 9
     }

--- a/FWCore/ServiceRegistry/README.md
+++ b/FWCore/ServiceRegistry/README.md
@@ -130,4 +130,3 @@ occur when accessing data from `edm::Ref`-style objects. |
 | `PreStreamEarlyTermination` | - | Emitted when an exception is thrown from any stream transition. |
 | `PreGlobalEarlyTermination` | - | Emitted when an exception is thrown from any global transition. |
 | `PreSourceEarlyTermination` | - | Emitted when an external termination request is received. |
-| `JobFailure` | - | To be removed (only called from `FWCore/Services/test/servicesJobReport_t.cpp`) |

--- a/FWCore/ServiceRegistry/README.md
+++ b/FWCore/ServiceRegistry/README.md
@@ -27,7 +27,6 @@ The order of the signals in this list is an approximate order in which they woul
 | `PostServicesConstruction` | - | Emitted after all services have been constructed. |
 | `{Pre,Post}EventSetupModulesConstruction` | Yes | Emitted before and after constructing EventSetup modules. |
 | `{Pre,Post}ESModuleConstruction` | Yes | Emitted before and after constructing an EventSetup module (ESProducer or ESSource). |
-| `PostESModuleRegistration` | - | Emitted after an ESModule provider has been registered with the EventSetupProvider. |
 | `{Pre,Post}ModulesAndSourceConstruction` | Yes | Emitted before and after concurrent construction of the Source and the EDModules. |
 | `{Pre,Post}SourceConstruction` | Yes | Emitted before and after construction of the Source. |
 | `{Pre,Post}OpenFile` | Yes | `PoolSource`, `EmbeddedRootSource`, `RNTupleTempSource`, and `EmbeddedRNTupleTempSource` emit these signals  before and after opening a new input file (either primary or secondary). Note that the first signal may be emitted during the Source construction, and the later signals after the `BeginProcessing`. |

--- a/FWCore/ServiceRegistry/interface/ActivityRegistry.h
+++ b/FWCore/ServiceRegistry/interface/ActivityRegistry.h
@@ -1297,13 +1297,6 @@ namespace edm {
     }
     AR_WATCH_USING_METHOD_1(watchPreSourceEarlyTermination)
 
-    /// signal is emitted if event processing or end-of-job
-    /// processing fails with an uncaught exception.
-    typedef signalslot::Signal<void()> JobFailure;
-    JobFailure jobFailureSignal_;
-    void watchJobFailure(JobFailure::slot_type const& iSlot) { jobFailureSignal_.connect_front(iSlot); }
-    AR_WATCH_USING_METHOD_0(watchJobFailure)
-
     // ---------- member functions ---------------------------
 
     ///forwards our signals to slots connected to iOther

--- a/FWCore/ServiceRegistry/interface/ActivityRegistry.h
+++ b/FWCore/ServiceRegistry/interface/ActivityRegistry.h
@@ -190,14 +190,6 @@ namespace edm {
     }
     AR_WATCH_USING_METHOD_1(watchPostESModuleConstruction)
 
-    /// signal is emitted after the ESModule is registered with EventSetupProvider
-    using PostESModuleRegistration = signalslot::Signal<void(eventsetup::ComponentDescription const&)>;
-    PostESModuleRegistration postESModuleRegistrationSignal_;
-    void watchPostESModuleRegistration(PostESModuleRegistration::slot_type const& iSlot) {
-      postESModuleRegistrationSignal_.connect(iSlot);
-    }
-    AR_WATCH_USING_METHOD_1(watchPostESModuleRegistration)
-
     using PreModulesAndSourceConstruction = signalslot::Signal<void()>;
     /// signal is emitted before the parallel section to construct ED modules and source
     PreModulesAndSourceConstruction preModulesAndSourceConstructionSignal_;

--- a/FWCore/ServiceRegistry/src/ActivityRegistry.cc
+++ b/FWCore/ServiceRegistry/src/ActivityRegistry.cc
@@ -222,7 +222,6 @@ namespace edm {
     preStreamEarlyTerminationSignal_.connect(std::cref(iOther.preStreamEarlyTerminationSignal_));
     preGlobalEarlyTerminationSignal_.connect(std::cref(iOther.preGlobalEarlyTerminationSignal_));
     preSourceEarlyTerminationSignal_.connect(std::cref(iOther.preSourceEarlyTerminationSignal_));
-    jobFailureSignal_.connect(std::cref(iOther.jobFailureSignal_));
   }
 
   void ActivityRegistry::copySlotsFrom(ActivityRegistry& iOther) {
@@ -395,6 +394,5 @@ namespace edm {
     copySlotsToFrom(preStreamEarlyTerminationSignal_, iOther.preStreamEarlyTerminationSignal_);
     copySlotsToFrom(preGlobalEarlyTerminationSignal_, iOther.preGlobalEarlyTerminationSignal_);
     copySlotsToFrom(preSourceEarlyTerminationSignal_, iOther.preSourceEarlyTerminationSignal_);
-    copySlotsToFromReverse(jobFailureSignal_, iOther.jobFailureSignal_);
   }
 }  // namespace edm

--- a/FWCore/ServiceRegistry/src/ActivityRegistry.cc
+++ b/FWCore/ServiceRegistry/src/ActivityRegistry.cc
@@ -60,7 +60,6 @@ namespace edm {
     postEventSetupModulesConstructionSignal_.connect(std::cref(iOther.postEventSetupModulesConstructionSignal_));
     preESModuleConstructionSignal_.connect(std::cref(iOther.preESModuleConstructionSignal_));
     postESModuleConstructionSignal_.connect(std::cref(iOther.postESModuleConstructionSignal_));
-    postESModuleRegistrationSignal_.connect(std::cref(iOther.postESModuleRegistrationSignal_));
     preModulesAndSourceConstructionSignal_.connect(std::cref(iOther.preModulesAndSourceConstructionSignal_));
     postModulesAndSourceConstructionSignal_.connect(std::cref(iOther.postModulesAndSourceConstructionSignal_));
     preSourceConstructionSignal_.connect(std::cref(iOther.preSourceConstructionSignal_));
@@ -231,7 +230,6 @@ namespace edm {
     copySlotsToFromReverse(postEventSetupModulesConstructionSignal_, iOther.postEventSetupModulesConstructionSignal_);
     copySlotsToFrom(preESModuleConstructionSignal_, iOther.preESModuleConstructionSignal_);
     copySlotsToFromReverse(postESModuleConstructionSignal_, iOther.postESModuleConstructionSignal_);
-    copySlotsToFromReverse(postESModuleRegistrationSignal_, iOther.postESModuleRegistrationSignal_);
     copySlotsToFrom(preModulesAndSourceConstructionSignal_, iOther.preModulesAndSourceConstructionSignal_);
     copySlotsToFromReverse(postModulesAndSourceConstructionSignal_, iOther.postModulesAndSourceConstructionSignal_);
     copySlotsToFrom(preSourceConstructionSignal_, iOther.preSourceConstructionSignal_);

--- a/FWCore/Services/plugins/StallMonitor.cc
+++ b/FWCore/Services/plugins/StallMonitor.cc
@@ -293,7 +293,7 @@ StallMonitor::StallMonitor(ParameterSet const& iPS, ActivityRegistry& iRegistry)
     iRegistry.watchPreModuleWriteLumi(this, &StallMonitor::preModuleGlobalTransition);
     iRegistry.watchPostModuleWriteLumi(this, &StallMonitor::postModuleGlobalTransition);
 
-    iRegistry.postESModuleRegistrationSignal_.connect([this](auto const& iDescription) {
+    iRegistry.postESModuleConstructionSignal_.connect([this](auto const& iDescription) {
       if (esModuleLabels_.size() <= iDescription.id_) {
         esModuleLabels_.resize(iDescription.id_ + 1);
       }

--- a/FWCore/Services/src/JobReportService.cc
+++ b/FWCore/Services/src/JobReportService.cc
@@ -19,7 +19,6 @@ namespace edm {
 
     JobReportService::JobReportService(ParameterSet const&, ActivityRegistry& reg) : JobReport() {
       reg.watchPostEndJob(this, &JobReportService::postEndJob);
-      reg.watchJobFailure(this, &JobReportService::frameworkShutdownOnFailure);
 
       // We don't handle PreProcessEvent, because we have to know *which
       // input file* was the event read from. Only the InputSource that
@@ -32,24 +31,9 @@ namespace edm {
 
       // ... not yet implemented ...
 
-      // Maybe we should have a member function called from both
-      // postEndJob() and frameworkShutdownOnFailure(), so that common
-      // elements are reported through common code.
-
       //
       // Any files that are still open should be flushed to the report
       //
-      impl()->flushFiles();
-    }
-
-    void JobReportService::frameworkShutdownOnFailure() {
-      // Dump information to the MessageLogger's JobSummary
-      // about the files that aren't already closed,
-      // and whatever summary information is wanted.
-
-      // Maybe we should have a member function called from both
-      // postEndJob() and frameworkShutdownOnFailure(), so that common
-      // elements are reported through common code.
       impl()->flushFiles();
     }
 

--- a/FWCore/Services/test/servicesJobReport_t.cpp
+++ b/FWCore/Services/test/servicesJobReport_t.cpp
@@ -59,25 +59,11 @@ void work()
   areg.postEndJobSignal_();
 }
 
-void fail()
-{
-  // Make the service.
-  edm::ParameterSet ps;
-  edm::ActivityRegistry areg;
-
-  edm::service::JobReportService jrs(ps, areg);
-  std::vector<edm::JobReport::Token> inputTokens;
-  std::vector<edm::JobReport::Token> outputTokens;
-
-  areg.jobFailureSignal_();
-}
-
 int main()
 {
   int rc = -1;
   try {
       work();
-      fail();
       rc = 0;
   }
 

--- a/FWCore/Services/test/servicesJobReport_t.cpp
+++ b/FWCore/Services/test/servicesJobReport_t.cpp
@@ -3,12 +3,10 @@
 #include <string>
 #include <vector>
 
-
 #include "FWCore/Utilities/interface/Exception.h"
 #include "FWCore/Services/src/JobReportService.h"
 
-void work()
-{
+void work() {
   // Make the service.
   edm::ParameterSet ps;
   edm::ActivityRegistry areg;
@@ -23,57 +21,47 @@ void work()
   branchnames.push_back("branch_2_");
 
   for (int i = 0; i < 10; ++i) {
-      std::ostringstream oss;
-      oss << i;
-      std::string seq_tag = oss.str();
-      std::vector<std::string> localBranchNames(branchnames);
-      localBranchNames[0] += seq_tag;
-      localBranchNames[1] += seq_tag;
-      edm::JobReport::Token t = jrs.inputFileOpened("phys" + seq_tag,
-				    "log" + seq_tag,
-				    "cat" + seq_tag,
-				    "class" + seq_tag,
-				    "label" + seq_tag,
-				    localBranchNames);
-      inputTokens.push_back(t);
+    std::ostringstream oss;
+    oss << i;
+    std::string seq_tag = oss.str();
+    std::vector<std::string> localBranchNames(branchnames);
+    localBranchNames[0] += seq_tag;
+    localBranchNames[1] += seq_tag;
+    edm::JobReport::Token t = jrs.inputFileOpened(
+        "phys" + seq_tag, "log" + seq_tag, "cat" + seq_tag, "class" + seq_tag, "label" + seq_tag, localBranchNames);
+    inputTokens.push_back(t);
   }
-
 
   jrs.reportSkippedEvent(10001, 1002);
   jrs.reportSkippedEvent(10001, 1003);
   jrs.reportSkippedEvent(10001, 1004);
 
   try {
-      jrs.eventReadFromFile(24, 0, 0);
-      assert( "Failed to throw required exception" == 0);
+    jrs.eventReadFromFile(24, 0, 0);
+    assert("Failed to throw required exception" == 0);
+  } catch (edm::Exception& x) {
+    assert(x.categoryCode() == edm::errors::LogicError);
+  } catch (...) {
+    assert("Threw unexpected exception type" == 0);
   }
-  catch ( edm::Exception & x ) {
-      assert( x.categoryCode() == edm::errors::LogicError );
-  }
-  catch ( ... ) {
-      assert( "Threw unexpected exception type" == 0 );
-  }
-  
 
   // Fake the end-of-job
   areg.postEndJobSignal_();
 }
 
-int main()
-{
+int main() {
   int rc = -1;
   try {
-      work();
-      rc = 0;
+    work();
+    rc = 0;
   }
 
-  catch ( edm::Exception & x ) {
-      std::cerr << x << '\n';
-      rc = 1;
-  }
-  catch ( ... ) {
-      std::cerr << "Unknown exception caught\n";
-      rc = 2;
+  catch (edm::Exception& x) {
+    std::cerr << x << '\n';
+    rc = 1;
+  } catch (...) {
+    std::cerr << "Unknown exception caught\n";
+    rc = 2;
   }
   return rc;
 }

--- a/PerfTools/AllocMonitor/plugins/ModuleAllocMonitor.cc
+++ b/PerfTools/AllocMonitor/plugins/ModuleAllocMonitor.cc
@@ -161,7 +161,7 @@ public:
         }
       });
 
-      iAR.watchPostESModuleRegistration([this](auto const& iDescription) {
+      iAR.watchPostESModuleConstruction([this](auto const& iDescription) {
         auto label = iDescription.label_;
         if (label.empty()) {
           label = iDescription.type_;

--- a/PerfTools/AllocMonitor/plugins/PhaseAllocMonitor.cc
+++ b/PerfTools/AllocMonitor/plugins/PhaseAllocMonitor.cc
@@ -145,7 +145,6 @@ public:
     REGISTER(PreCloseOutputFiles);
     REGISTER(PostCloseOutputFiles);
 
-    REGISTER(JobFailure);
     REGISTER(BeginProcessing);
     REGISTER(EndProcessing);
 

